lib/views/components/loading_icon_button.dart

```diff
@@ -0,0 +1,71 @@
+import "package:flutter/material.dart";
+import "package:flutter_hooks/flutter_hooks.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+
+typedef FutureVoidCallback = Future<void> Function();
```

Comment:
Good idea to define a `typedef` !
I was just wondering if this file was a good place to define it.
May be we should have it in a utility file instead ?

Reply:
Yes I see.
I believe we can still add a util file even if it is one line because we might forget to refactor the current implementation in the future I feel like.

Reply:
ok done in 23133e5